content/blog/stringr-1-5-0/index.Rmd

```diff
@@ -0,0 +1,259 @@
+-->
+```
+We're chuffed to announce the release of [stringr](https://stringr.tidyverse.org) 1.5.0.
+stringr provides a cohesive set of functions designed to make working with strings as easy as possible.
+
+You can install it from CRAN with:
+
+```{r, eval = FALSE}
+install.packages("stringr")
+```
+
+This blog post will give you an overview of the biggest changes (you can get a detailed list of all changes from the [release notes](%7B%20github_release%20%7D)).
+Firstly, we need to update you on some (small) breaking changes we've made to make stringr more consistent with the rest of the tidyverse.
+Then we'll give a quick overview of improvement to documentation and stringr's new license.
+Then we'll finish off by diving into the many small, but useful, functions that we've accumulated in the three and half years since the last release.
+
+```{r setup}
+library(stringr)
+```
+
+## Breaking changes
+
+Lets start with the important stuff: the breaking changes.
+We've tried to keep these small and we don't believe they'll affect much code in the wild (only affected \~20 of the \~1600 packages that use stringr).
+But we're believe they're important as a consistent set of rules makes the tidyverse as a whole more predictable and easier to learn.
+
+### Recycling rules
+
+stringr functions now consistently implement the tidyverse recycling rules[^1], which are stricter than the previous rules in two ways.
+Firstly, we no longer recycle shorter vectors that are an integer multiple of longer vectors:
+
+[^1]: You might wonder why we developed our own set of recycling rules for the tidyverse instead of using the base R rules.
+    That's because, unfortunately, there isn't a consistent set of rules used by base R, but a [suite of variations](https://vctrs.r-lib.org/articles/type-size.html#appendix-recycling-in-base-r).
+
+```{r}
+#| error: true
+
+str_detect(letters, c("x", "y"))
+```
+
+Secondly, a 0-length vector no longer implies a 0-length output.
+Instead it's recycled using the usual rules:
+
+```{r}
+#| error: true
+
+str_detect(letters, character())
+str_detect("x", character())
+```
+
+Neither of these situations occurs very commonly in data analysis, so this change primarily brings consistency with the rest of the tidyverse without affecting much existing code.
+
+Finally, stringr functions are generally a little stricter because we require the inputs to be vectors of some type.
+Again, this is unlikely to affect your data analysis code and will result in a clearer error if you accidentally pass in something weird:
+
+```{r}
+#| error: TRUE
+
+str_detect(mean, "x")
+```
+
+### Empty patterns
+
+In many stringr functions, `""` will match or split on every character motivated.
+This is motivated by base R's `strsplit()`:
+
+```{r}
+strsplit("abc", "")[[1]]
+str_split("abc", "")[[1]]
+```
+
+When creating stringr (over 13 years ago!) I took this idea and ran with it, implementing similar support in every function where it might possibly work.
```

Comment:
```suggestion
When creating stringr (over 13 years ago!), I took this idea and ran with it, implementing similar support in every function where it might possibly work.
```